Lib/calendar.py

```diff
@@ -565,8 +577,6 @@ class LocaleTextCalendar(TextCalendar):
 
     def __init__(self, firstweekday=0, locale=None):
         TextCalendar.__init__(self, firstweekday)
-        if locale is None:
-            locale = _locale.getlocale(_locale.LC_TIME)
         self.locale = locale
```

Comment:
Suggestion:

```python
if locale is None:
    locale = _locale.getlocale(_locale.LC_TIME)
    if locale == (None, None):
        with different_locale(""):
            locale = _locale.getlocale(_locale.LC_TIME)
    self.locale = locale
```

Reply:
I would prefer to not change the locale in the constructor. That's why I proposed this specific PR.

Reply:
Why is not okay to temporarily modify `LC_TIME` in `__init__()`, but fine to do so in `formatweekday()` and `formatmonthname()`? The class is documented as not thread safe with respect to the process locale.

Reply:
It is ok to call setlocale() in ``__init__()``. But setlocale() affects all threads, so I prefer to avoid calling it if possible.

Reply:
I think that change is the less of the two. I don't think it's worth changing the behavior from the previous code that looked up the default locale to use via `getdefaultlocale()` only once, instead of calling it on every entry of `different_locale`. But that's just me voicing one opinion. I'd prefer for other people to way on exactly which default behaviors are important to preserve, or whether the *entire default behavior* of `locale=None` should change to use the `LC_TIME` locale even if it's "C".